qemu/tests/qsd_backup_push.py

```diff
@@ -0,0 +1,152 @@
+        error_context.context("Create target disk")
+        for tag in self.params.objects("source_images"):
+            image_params = self.params.object_params(tag)
+            for img in image_params.objects("image_backup_chain"):
+                disk = self.target_disk_define_by_params(self.params, img)
+                disk.hotplug(self.qsd)
+                self.trash.append(disk)
+
+    def do_full_backup(self):
+        extra_options = {"sync": "full", "auto_disable_bitmap": False}
+        backup_utils.blockdev_batch_backup(
+            self.qsd,
+            self.source_images,
+            self.full_backups,
+            self.bitmaps,
+            **extra_options)
+
+    def generate_inc_files(self):
+        for tag in self.params.objects("source_images"):
+            self.generate_data_file(tag)
+
+    def do_increamental_backup(self):
+        extra_options = {'sync': 'incremental', 'auto_disable_bitmap': False}
+        backup_utils.blockdev_batch_backup(
+            self.qsd,
+            self.source_images,
+            self.inc_backups,
+            self.bitmaps,
+            **extra_options)
+
+    def rebase_target_disk(self):
+        self.qsd.stop_daemon()
+        return utils_misc.parallel(self.rebase_targets)
+
+    def prepare_clone_vm(self):
+        self.main_vm.destroy()
+        images = self.params["images"]
+        clone_params = self.main_vm.params.copy()
+        for tag in self.params.objects("source_images"):
+            img_params = self.params.object_params(tag)
+            image_chain = img_params.objects('image_backup_chain')
+            images = images.replace(tag, image_chain[-1])
+            self.params["qsd_images_qsd1"] = image_chain[-1]
+        clone_params["images"] = images
+        clone_vm = self.main_vm.clone(params=clone_params)
+        self.start_qsd()
+        clone_vm.create()
+        clone_vm.verify_alive()
+        self.clone_vm = clone_vm
+
+    def do_test(self):
+        self.do_full_backup()
+        self.generate_inc_files()
+        self.do_increamental_backup()
+        self.main_vm.destroy()
+        self.rebase_target_disk()
+        memory.drop_caches()
+        self.verify_target_disk()
+
+    def verify_target_disk(self):
+        self.prepare_clone_vm()
+        try:
+            self.verify_data_files()
+        finally:
+            self.clone_vm.destroy()
+
+    def prepare_test(self):
+        self.start_qsd()
+        super(QSDBackupTest, self).prepare_test()
+
+    def post_test(self):
+        super(QSDBackupTest, self).post_test()
+        self.qsd.stop_daemon()
+
+
+def run(test, params, env):
+    """
+    increamental backup test via qsd
+     test steps:
+        1. export data disk via qsd+nbd.
+        2. boot VM with the exported data disk
+        3. make filesystem in data disks
+        4. create file and save it md5sum in data disks
+        5. add target disks for backup to VM via qmp commands
```

Comment:
hotplug backup images to qsd via qmp commands

Reply:
Your steps description here is still OK for me, so you can ignore these comments.

I added comments here in order to add a note that if we can make it clear to whoever never know the background of block backup.